src/Symfony/Contracts/EventDispatcher/EventDispatcherInterface.php

```diff
@@ -13,46 +13,23 @@
 
 use Psr\EventDispatcher\EventDispatcherInterface as PsrEventDispatcherInterface;
 
-if (interface_exists(PsrEventDispatcherInterface::class)) {
-    /**
-     * Allows providing hooks on domain-specific lifecycles by dispatching events.
-     */
-    interface EventDispatcherInterface extends PsrEventDispatcherInterface
-    {
-        /**
-         * Dispatches an event to all registered listeners.
-         *
-         * For BC with Symfony 4, the $eventName argument is not declared explicitly on the
-         * signature of the method. Implementations that are not bound by this BC constraint
-         * MUST declare it explicitly, as allowed by PHP.
-         *
-         * @param object      $event     The event to pass to the event handlers/listeners
-         * @param string|null $eventName The name of the event to dispatch. If not supplied,
-         *                               the class of $event should be used instead.
-         *
-         * @return object The passed $event MUST be returned
-         */
-        public function dispatch($event/*, string $eventName = null*/);
-    }
-} else {
+/**
+ * Allows providing hooks on domain-specific lifecycles by dispatching events.
+ */
+interface EventDispatcherInterface extends PsrEventDispatcherInterface
+{
     /**
-     * Allows providing hooks on domain-specific lifecycles by dispatching events.
+     * Dispatches an event to all registered listeners.
+     *
+     * For BC with Symfony 4, the $eventName argument is not declared explicitly on the
+     * signature of the method. Implementations that are not bound by this BC constraint
+     * MUST declare it explicitly, as allowed by PHP.
+     *
+     * @param object      $event     The event to pass to the event handlers/listeners
+     * @param string|null $eventName The name of the event to dispatch. If not supplied,
+     *                               the class of $event should be used instead.
+     *
+     * @return object The passed $event MUST be returned
      */
-    interface EventDispatcherInterface
-    {
-        /**
-         * Dispatches an event to all registered listeners.
-         *
-         * For BC with Symfony 4, the $eventName argument is not declared explicitly on the
-         * signature of the method. Implementations that are not bound by this BC constraint
-         * MUST declare it explicitly, as allowed by PHP.
-         *
-         * @param object      $event     The event to pass to the event handlers/listeners
-         * @param string|null $eventName The name of the event to dispatch. If not supplied,
-         *                               the class of $event should be used instead.
-         *
-         * @return object The passed $event MUST be returned
-         */
-        public function dispatch($event/*, string $eventName = null*/);
-    }
+    public function dispatch(object $event/*, string $eventName = null*/);
```

Comment:
can't we add the argument here?

Reply:
ok, added

Reply:
EventDispatcher 4.4 would become incompatible in that case.

https://github.com/symfony/symfony/blob/a676feb66b3629da72375eb985f2ecd0e065116d/src/Symfony/Component/EventDispatcher/EventDispatcher.php#L51

Reply:
Not really an issue, isn't it?

Reply:
Staying compatible with 4.4 implementations was the idea of contracts v2, IIRC.

Reply:
I understand - yet there are no other implementations of the contracts as of now.
Better break "hard" now that nobody is affected, no?

Reply:
🤔 I guess you're right. Let's keep the parameter for now since I cannot really come up with a good reason against it. 😃